Perform bert algorithm model distillation on msra data set based on ofa architecture

AI-Mart opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/PaddleNLP/pull/26 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->

### PR changes
<!-- One of [ Models | APIs | Docs | Others ] -->

### Description
<!-- Describe what this PR does -->

AI-Mart committed:
Update train.py

AI-Mart committed:
基于ofa架构在msra数据集上进行bert算法模型蒸馏

基于ofa架构在msra数据集上进行bert算法模型蒸馏

AI-Mart commented:
Perform bert algorithm model distillation on msra data set based on ofa architecture

AI-Mart reviewed (commented):
Perform bert algorithm model distillation on msra data set based on ofa architecture

LiuChiachi commented on examples/model_compression/ofa/README_MSRA.md:
```diff
@@ -0,0 +1,93 @@
+# 使用PaddleNLP一键启动，完成基于内置数据集msra的中文命名实体识别模型工业化落地训练，并进行模型压缩适用于工业化落地场景
```
一级标题是不是可以简短一点。这句话可能不适合作为该样例的标题。

AI-Mart replied:
已调整

LiuChiachi commented on examples/model_compression/ofa/export_model_msra_ner_ofa.py:
```diff
@@ -0,0 +1,239 @@
+        if isinstance(sublayer, paddle.nn.MultiHeadAttention):
+            sublayer.num_heads = int(args.width_mult * sublayer.num_heads)
+
+    # output_dir = os.path.join(args.sub_model_output_dir,
```
可以删去无意义的注释

AI-Mart replied:
已调整

LiuChiachi commented on examples/model_compression/ofa/requirements.txt:
```diff
@@ -0,0 +1,3 @@
+paddlepaddle==2.2.0
+paddlenlp==2.1.1
```
这里是否需要加入Paddle和paddlenlp的环境依赖呢

AI-Mart replied:
不需要的，安装这几个主要包，依赖会自动安装，试验过，附上依赖可能会存在安装冲突

ZeyuChen replied:
删除这个文件吧，我理解使用这个exmaples的开发者应该都是有了。限制住这些版本会导致未来升级出现问题。
只需要文档里特殊注明需要额外依赖paddleslim即可

LiuChiachi commented on examples/model_compression/ofa/run_msra_ner.py:
```diff
@@ -0,0 +1,229 @@
+from paddlenlp.metrics import ChunkEvaluator
+from paddlenlp.datasets import load_dataset
+from paddlenlp.transformers import BertForTokenClassification, BertTokenizer
+from paddlenlp.transformers import ErnieCtmForTokenClassification, ErnieCtmTokenizer
```
没有用到的import可以删掉，包括下面无意义的注释。

AI-Mart replied:
已调整

LiuChiachi commented on examples/model_compression/ofa/run_msra_ner.py:
```diff
@@ -0,0 +1,229 @@
+        'msra_ner', splits=('train', 'test'), lazy=False)
+
+    tokenizer = BertTokenizer.from_pretrained(args.model_name_or_path)
+    # tokenizer = ErnieCtmTokenizer.from_pretrained("ernie-ctm")
```
可以删掉无意义的注释

AI-Mart replied:
已调整

LiuChiachi commented on examples/model_compression/ofa/run_msra_ner.py:
```diff
@@ -0,0 +1,229 @@
+    model = BertForTokenClassification.from_pretrained(
+        args.model_name_or_path, num_classes=label_num)
+    # model = ErnieCtmForTokenClassification.from_pretrained(
+    #    args.model_name_or_path, num_classes=label_num)
```
可以删掉无意义的注释

AI-Mart replied:
已调整

LiuChiachi commented on examples/model_compression/ofa/run_msra_ner_ofa.py:
```diff
@@ -0,0 +1,662 @@
+                fp.write(
+                    'epoch:%s, step: %s, width_mult: %s, eval loss: %f, mcc: %s, \n'
+                    % (epoch, global_step, str(width_mult), eval_loss, res[0]))
+        elif isinstance(metric, PearsonAndSpearman):
```
这些分支是否都需要呢，如果不需要可以都删掉。

AI-Mart replied:
已调整

LiuChiachi commented on examples/model_compression/ofa/README_MSRA.md:
```diff
@@ -0,0 +1,93 @@
+- `output_dir` 表示模型保存路径。
+- `device` 表示训练使用的设备, 'gpu'表示使用GPU, 'xpu'表示使用百度昆仑卡, 'cpu'表示使用CPU。
+- `width_mult_list` 表示压缩训练过程中，对每层Transformer Block的宽度选择的范围。
+
```
是否可以补充出详细的精度数据，比如压缩前多少，压缩的百分比，以及压缩之后的精度是多少。

AI-Mart replied:
已调整

LiuChiachi commented on examples/model_compression/ofa/README_MSRA.md:
```diff
@@ -0,0 +1,93 @@
+- `static_sub_model` 指示了导出子模型静态图模型及参数的目录，设置为None，则表示不导出静态图模型。默认：None。
+- `device` 表示训练使用的设备, 'gpu'表示使用GPU, 'xpu'表示使用百度昆仑卡, 'cpu'表示使用CPU。
+- `width_mult` 表示导出子模型的宽度。默认：1.0.
+
```
如果导出子模型后进行预测了，是否可以提供出与教师模型相比，推理加速的百分比呢？

AI-Mart replied:
已调整

AI-Mart committed:
Add files via upload

AI-Mart commented:
已经按要求更改，并上传

ZeyuChen commented on examples/model_compression/ofa/README_MSRA.md:
```diff
@@ -0,0 +1,102 @@
+# 基于ofa架构在msra实体识别任务的模型压缩
```
这一README是否可以与主README合并。

AI-Mart replied:
已经调整

AI-Mart committed:
Delete README_MSRA.md

AI-Mart committed:
Delete requirements.txt

AI-Mart committed:
Add files via upload

AI-Mart commented:
已经按要求调整

AI-Mart committed:
2 bug in write_prediction_results of utils file The .zip file must save the above files first, and then save the zip file, otherwise each .zip is the result of the previous prediction 3 The bug of get_precision_recall_f1 of utils file When f1 is 1.0, the match is 0, and it is changed to regular direct matching. 4 Avoid hard coding and make fixed values into macro definitions for easy porting （修改后）恢复原文

AI-Mart committed:
utils bugs

1 bug in write_prediction_results of utils file
The .zip file must save the above files first, and then save the zip file, otherwise each .zip is the result of the previous prediction
2 The bug of get_precision_recall_f1 of utils file
When f1 is 1.0, the match is 0, and it is changed to regular direct matching.
3 Avoid hard coding and make fixed values into macro definitions for easy porting

ZeyuChen commented:
@AI-Mart DuIE的部分PR请分开提交。我们尽量提交小PR方便review
按功能拆分一个个合入会比较好，否则现在代码出现冲突了，看是否可以拿掉DuIE部分的PR合入。

AI-Mart committed:
Add files via upload

AI-Mart committed:
Delete data_loader.py

AI-Mart committed:
Delete utils.py

AI-Mart commented:
已经删除

AI-Mart committed:
Add files via upload

AI-Mart committed:
Add files via upload

AI-Mart committed:
Delete examples/information_extraction/DuIE directory

AI-Mart committed:
Delete examples/information_extraction/msra_ner directory

AI-Mart commented:
因为合并导致冲突，我重新创建了个PR 地址：https://github.com/PaddlePaddle/PaddleNLP/pull/1598，这个先关了。
